notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaNotificationRESTService.java

```diff
@@ -34,72 +37,103 @@
  * @author Josh Helmer, jhelmer.unicon.net
  * @since 3.0
  */
-@Service
+@Service("jpaNotificationRestService")
 public class JpaNotificationRESTService implements IJpaNotificationRESTService {
+
     @Autowired
     private INotificationDao notificationDao;
+
+    @Autowired
+    private JpaNotificationService jpaNotificationService;
+
     @Autowired
     private INotificationDTOMapper notificationMapper;
 
+    private Logger logger = LoggerFactory.getLogger(getClass());
 
     @Override
     @Transactional(readOnly = true)
-    public List<EntryDTO> getNotifications(Integer page, Integer pageSize) {
-        List<JpaEntry> entries = notificationDao.list(page, pageSize);
+    public EntryDTO getNotification(long id, boolean full) {
+        final JpaEntry entry = full ? notificationDao.getFullEntry(id) : notificationDao.getEntry(id);
+        if (entry == null) {
+            return null;
+        }
 
-        return notificationMapper.toEntryList(entries);
+        return notificationMapper.toEntry(entry);
     }
 
-
     @Override
     @Transactional(readOnly = true)
-    public EntryDTO getNotification(long id, boolean full) {
-        JpaEntry entry = full ? notificationDao.getFullEntry(id) : notificationDao.getEntry(id);
-        if (entry == null) {
-            return null;
+    public EntryDTO getNotification(NotificationEntry entry, boolean full) {
+        EntryDTO rslt = null;  // default
+        if (jpaNotificationService.contains(entry)) {
+            final String idString = entry.getId().substring(JpaNotificationService.ID_PREFIX.length());
+            final long id = Long.parseLong(idString);
+            rslt = getNotification(id, full);
         }
+        return rslt;
+    }
 
-        return notificationMapper.toEntry(entry);
+    @Override
+    @Transactional(readOnly = true)
+    public List<EntryDTO> getNotifications(Integer page, Integer pageSize) {
+        final List<JpaEntry> entries = notificationDao.list(page, pageSize);
+
+        return notificationMapper.toEntryList(entries);
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    public List<EntryDTO> getNotificationsBySourceAndCustomAttribute(String source, String attributeName, String attributeValue) {
+        Validate.notBlank(source, "Argument 'source' cannot be blank");
+        Validate.notBlank(source, "Argument 'attributeName' cannot be blank");
```

Comment:
Should line 89 be
`Validate.notBlank(attributeName, "Argument 'attributeName' cannot be blank");`?